### ISSUE TITLE
Deploy functions not working

### ISSUE THREAD

**Issue by arthurgubaidullin** [firebase/firebase-tools#4735]
### [REQUIRED] Environment info

**firebase-tools:** 11.2.2

**Platform:** macOS

### [REQUIRED] Test case

`firebase deploy --only functions`

### [REQUIRED] Steps to reproduce

`firebase deploy --only functions`

### [REQUIRED] Expected behavior

Must works.

### [REQUIRED] Actual behavior

Throw error: `Error: CEL evaluation of expression 'undefined' not yet supported`

NDA.

**Comment by bkendall** [comment]
Deploying functions from my test projects seems to be working just fine. There's not enough information here to debug your issues. Can you provide a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that shows this error?

**Comment by ericvera** [comment]
I received the same error when starting the functions emulator with 12.2.2. Works fine with 12.2.1.

**Comment by taeold** [comment]
@ericvera Do you mind sharing some of your code that's surfacing these errors. I'm having some trouble trying to reproduce the issue.

**Comment by r-dev-limited** [comment]
I can see same error with emulator

**Comment by ianitsky** [comment]
I received the following error:



[debug] [2022-07-15T15:27:12.832Z] <<< [apiv2][body] PATCH https://cloudfunctions.googleapis.com/v1/projects/*/locations/us-central1/functions/* {"error":{"code":400,"message":"The request has errors","status":"INVALID_ARGUMENT","details":[{"@type":"type.googleapis.com/google.rpc.BadRequest","fieldViolations":[{"field":"vpc_connector","description":"A Cloud Function and a Serverless VPC Access connector must be located in the same region."}]}]}}

But the function and the VPC connector are on the same region.

The upload works fine on version 11.1.0

**Comment by bkendall** [comment]
@ianitsky could you open a separate issue? This is going to be tracking the `"CEL evaluation"` issue, which I don't think you're experiencing. Thanks

**Comment by arthurgubaidullin** [comment]
`firebase-tools` `v11.3.0` also affected. `v11.1.0` not affected.

**Comment by micro-luke** [comment]
I see this issue as well - for me it only occurs when I include a "task" function in the deployment (just started experimenting with it). If I comment out the task function everything is deploying.
firebase-tools - 11.3.0
firebase-admin - 10.3.0

**Comment by radvansky-tomas** [comment]
Same error if you try to run emulator suite

```
> firebase emulators:exec --inspect-functions --ui --only functions,firestore,auth,pubsub,storage "npm run watch"

i  emulators: Starting emulators: auth, functions, firestore, pubsub, storage
⚠  functions: You are running the Functions emulator in debug mode (port=9229). This means that functions will execute in sequence rather than in parallel.
⚠  functions: The following emulators are not running, calls to these services from the Functions emulator will affect production: database, hosting
✔  functions: Using node@16 from host.
i  firestore: Firestore Emulator logging to firestore-debug.log
i  pubsub: Pub/Sub Emulator logging to pubsub-debug.log
i  ui: Emulator UI logging to ui-debug.log
i  functions: Watching "/Users/tomasradvansky/Documents/RDEV-LIMITED/Blockparty/block-party/functions" for Cloud Functions...
i  emulators: Shutting down emulators.
i  ui: Stopping Emulator UI
⚠  Emulator UI has exited upon receiving signal: SIGINT
i  functions: Stopping Functions Emulator
i  firestore: Stopping Firestore Emulator
i  pubsub: Stopping Pub/Sub Emulator
i  auth: Stopping Authentication Emulator
i  storage: Stopping Storage Emulator
i  hub: Stopping emulator hub
i  logging: Stopping Logging Emulator

Error: CEL evaluation of expression 'undefined' not yet supported
```

**Comment by tojtoj** [comment]
> Having the same issue here when trying to deploy a task function (first time deploying a task function):
> `Error: CEL evaluation of expression 'undefined' not yet supported`

Same here, downgrade to 11.1.0 helped.

**Comment by colerogers** [comment]
We just released firebase-tools `11.4.2` that should fix this issue. Please upgrade to that version and retry task function deployments. Thanks

**Comment by arthurgubaidullin** [comment]
@colerogers  Thanks. I'll check tomorrow at work.

**Comment by janm6k** [comment]
It fixed the task functions for my project, thanks!

**Comment by arthurgubaidullin** [comment]
It fixed. Deployment works! Thanks.

**Comment by micro-luke** [comment]
Great stuff - issue fixed for me also.
